api_schema/agent_config.go

```diff
@@ -0,0 +1,18 @@
+package api_schema
+
+import "regexp"
+
+// Reflects the version of the JSON encoding of ServiceAgentConfig. Increase the
+// minor version number for backwards-compatible changes, and the major version
+// number for non-backwards-compatible changes.
+const ServiceAgentConfigVersion = "v0.1"
+
+type ServiceAgentConfig struct {
+	FormatVersion  string               `json:"version"`
+	FieldsToRedact FieldRedactionConfig `json:"fields_to_redact"`
+}
+
+type FieldRedactionConfig struct {
+	FieldNames       []string         `json:"field_names"`
+	FieldNameRegexps []*regexp.Regexp `json:"field_name_regexps"`
```

Comment:
Huh, TIL that regexp.Regexp implements UnmarshalText!

Reply:
Yup, we even get a JSON deserialization error if we try to deserialize an invalid regular expression!